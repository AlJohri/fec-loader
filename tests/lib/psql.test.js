const { ObjectReadableMock, ObjectWritableMock } = require('stream-mock');
const { advanceTo } = require('jest-date-mock');

const psql = require('../../lib/psql');
const fs = require('fs');

test('psql', async done => {
    const input = fs.createReadStream(__dirname + '/../fixtures/989514.fec');
    const writer = new ObjectWritableMock();

    advanceTo(new Date(2019, 10, 5, 0, 0, 0));

    psql({
        in: input,
        out: writer
    });

    writer.on('finish', () => {
        expect(writer.data).toMatchSnapshot();
        done();
    });
});
