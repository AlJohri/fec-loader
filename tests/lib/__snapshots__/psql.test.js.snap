// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`psql 1`] = `
Array [
  "COPY fec_filings (\\"record_type\\", \\"ef_type\\", \\"fec_version\\", \\"soft_name\\", \\"soft_ver\\", \\"report_id\\", \\"report_number\\", \\"comment\\", \\"filing_id\\", \\"created_date\\", \\"updated_date\\") FROM STDIN;
",
  "HDR	FEC	8.1	CMDI CRIMSON FILER	2.0	\\\\N	\\\\N	\\\\N	\\\\N	2019-11-05T05:00:00.000Z	2019-11-05T05:00:00.000Z
",
  "\\\\.
",
  "COPY fec_presidential_summaries (\\"form_type\\", \\"filer_committee_id_number\\", \\"committee_name\\", \\"change_of_address\\", \\"street_1\\", \\"street_2\\", \\"city\\", \\"state\\", \\"zip_code\\", \\"activity_primary\\", \\"activity_general\\", \\"report_code\\", \\"election_code\\", \\"date_of_election\\", \\"state_of_election\\", \\"coverage_from_date\\", \\"coverage_through_date\\", \\"treasurer_last_name\\", \\"treasurer_first_name\\", \\"treasurer_middle_name\\", \\"treasurer_prefix\\", \\"treasurer_suffix\\", \\"date_signed\\", \\"col_a_cash_on_hand_beginning_period\\", \\"col_a_total_receipts\\", \\"col_a_subtotal\\", \\"col_a_total_disbursements\\", \\"col_a_cash_on_hand_close_of_period\\", \\"col_a_debts_to\\", \\"col_a_debts_by\\", \\"col_a_expenditures_subject_to_limits\\", \\"col_a_net_contributions\\", \\"col_a_net_operating_expenditures\\", \\"col_a_federal_funds\\", \\"col_a_individuals_itemized\\", \\"col_a_individuals_unitemized\\", \\"col_a_individual_contribution_total\\", \\"col_a_political_party_committees_receipts\\", \\"col_a_other_political_committees_pacs\\", \\"col_a_the_candidate\\", \\"col_a_total_contributions\\", \\"col_a_transfers_from_aff_other_party_cmttees\\", \\"col_a_received_from_or_guaranteed_by_cand\\", \\"col_a_other_loans\\", \\"col_a_total_loans\\", \\"col_a_operating\\", \\"col_a_fundraising\\", \\"col_a_legal_and_accounting\\", \\"col_a_total_offsets_to_expenditures\\", \\"col_a_other_receipts\\", \\"col_a_operating_expenditures\\", \\"col_a_transfers_to_other_authorized_committees\\", \\"col_a_fundraising_disbursements\\", \\"col_a_exempt_legal_accounting_disbursement\\", \\"col_a_made_or_guaranteed_by_candidate\\", \\"col_a_other_repayments\\", \\"col_a_total_loan_repayments_made\\", \\"col_a_individuals\\", \\"col_a_political_party_committees_refunds\\", \\"col_a_other_political_committees\\", \\"col_a_total_contributions_refunds\\", \\"col_a_other_disbursements\\", \\"col_a_items_on_hand_to_be_liquidated\\", \\"col_a_alabama\\", \\"col_a_alaska\\", \\"col_a_arizona\\", \\"col_a_arkansas\\", \\"col_a_california\\", \\"col_a_colorado\\", \\"col_a_connecticut\\", \\"col_a_delaware\\", \\"col_a_dist_of_columbia\\", \\"col_a_florida\\", \\"col_a_georgia\\", \\"col_a_hawaii\\", \\"col_a_idaho\\", \\"col_a_illinois\\", \\"col_a_indiana\\", \\"col_a_iowa\\", \\"col_a_kansas\\", \\"col_a_kentucky\\", \\"col_a_louisiana\\", \\"col_a_maine\\", \\"col_a_maryland\\", \\"col_a_massachusetts\\", \\"col_a_michigan\\", \\"col_a_minnesota\\", \\"col_a_mississippi\\", \\"col_a_missouri\\", \\"col_a_montana\\", \\"col_a_nebraska\\", \\"col_a_nevada\\", \\"col_a_new_hampshire\\", \\"col_a_new_jersey\\", \\"col_a_new_mexico\\", \\"col_a_new_york\\", \\"col_a_north_carolina\\", \\"col_a_north_dakota\\", \\"col_a_ohio\\", \\"col_a_oklahoma\\", \\"col_a_oregon\\", \\"col_a_pennsylvania\\", \\"col_a_rhode_island\\", \\"col_a_south_carolina\\", \\"col_a_south_dakota\\", \\"col_a_tennessee\\", \\"col_a_texas\\", \\"col_a_utah\\", \\"col_a_vermont\\", \\"col_a_virginia\\", \\"col_a_washington\\", \\"col_a_west_virginia\\", \\"col_a_wisconsin\\", \\"col_a_wyoming\\", \\"col_a_puerto_rico\\", \\"col_a_guam\\", \\"col_a_virgin_islands\\", \\"col_a_totals\\", \\"col_b_federal_funds\\", \\"col_b_individuals_itemized\\", \\"col_b_individuals_unitemized\\", \\"col_b_individual_contribution_total\\", \\"col_b_political_party_committees_receipts\\", \\"col_b_other_political_committees_pacs\\", \\"col_b_the_candidate\\", \\"col_b_total_contributions_other_than_loans\\", \\"col_b_transfers_from_aff_other_party_cmttees\\", \\"col_b_received_from_or_guaranteed_by_cand\\", \\"col_b_other_loans\\", \\"col_b_total_loans\\", \\"col_b_operating\\", \\"col_b_fundraising\\", \\"col_b_legal_and_accounting\\", \\"col_b_total_offsets_to_operating_expenditures\\", \\"col_b_other_receipts\\", \\"col_b_total_receipts\\", \\"col_b_operating_expenditures\\", \\"col_b_transfers_to_other_authorized_committees\\", \\"col_b_fundraising_disbursements\\", \\"col_b_exempt_legal_accounting_disbursement\\", \\"col_b_made_or_guaranteed_by_the_candidate\\", \\"col_b_other_repayments\\", \\"col_b_total_loan_repayments_made\\", \\"col_b_individuals\\", \\"col_b_political_party_committees_refunds\\", \\"col_b_other_political_committees\\", \\"col_b_total_contributions_refunds\\", \\"col_b_other_disbursements\\", \\"col_b_total_disbursements\\", \\"col_b_alabama\\", \\"col_b_alaska\\", \\"col_b_arizona\\", \\"col_b_arkansas\\", \\"col_b_california\\", \\"col_b_colorado\\", \\"col_b_connecticut\\", \\"col_b_delaware\\", \\"col_b_dist_of_columbia\\", \\"col_b_florida\\", \\"col_b_georgia\\", \\"col_b_hawaii\\", \\"col_b_idaho\\", \\"col_b_illinois\\", \\"col_b_indiana\\", \\"col_b_iowa\\", \\"col_b_kansas\\", \\"col_b_kentucky\\", \\"col_b_louisiana\\", \\"col_b_maine\\", \\"col_b_maryland\\", \\"col_b_massachusetts\\", \\"col_b_michigan\\", \\"col_b_minnesota\\", \\"col_b_mississippi\\", \\"col_b_missouri\\", \\"col_b_montana\\", \\"col_b_nebraska\\", \\"col_b_nevada\\", \\"col_b_new_hampshire\\", \\"col_b_new_jersey\\", \\"col_b_new_mexico\\", \\"col_b_new_york\\", \\"col_b_north_carolina\\", \\"col_b_north_dakota\\", \\"col_b_ohio\\", \\"col_b_oklahoma\\", \\"col_b_oregon\\", \\"col_b_pennsylvania\\", \\"col_b_rhode_island\\", \\"col_b_south_carolina\\", \\"col_b_south_dakota\\", \\"col_b_tennessee\\", \\"col_b_texas\\", \\"col_b_utah\\", \\"col_b_vermont\\", \\"col_b_virginia\\", \\"col_b_washington\\", \\"col_b_west_virginia\\", \\"col_b_wisconsin\\", \\"col_b_wyoming\\", \\"col_b_puerto_rico\\", \\"col_b_guam\\", \\"col_b_virgin_islands\\", \\"col_b_totals\\", \\"filing_id\\") FROM STDIN;
",
  "F3PN	C00431171	ROMNEY FOR PRESIDENT, INC.	\\\\N	C/O RED CURVE SOLUTIONS, LLC	500 CUMMINGS CENTER, SUITE 4400	BEVERLY	MA	01915	\\\\N	\\\\N	YE	\\\\N	\\\\N	\\\\N	20141125	20141231	Crate	Darrell	\\\\N	\\\\N	\\\\N	20150130	2176183.62	12592.00	2188775.62	157829.70	2030945.92	0.00	0.00	0.00	-617988.26	16330140.35	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	12592.00	157829.70	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	440290.98	214971.78	655262.76	0.00	38713.49	0.00	693976.25	3490846.12	0.00	0.00	0.00	7285341.61	0.00	0.00	7285341.61	3598579.11	15068743.09	23615481.96	444686.59	0.00	0.00	0.00	1500000.00	1500000.00	1281800.02	0.00	30164.49	1311964.51	184865.62	27056998.68	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	0.00	\\\\N
",
  "\\\\.
",
  "COPY fec_contributions (\\"form_type\\", \\"filer_committee_id_number\\", \\"transaction_id\\", \\"back_reference_tran_id_number\\", \\"back_reference_sched_name\\", \\"entity_type\\", \\"contributor_organization_name\\", \\"contributor_last_name\\", \\"contributor_first_name\\", \\"contributor_middle_name\\", \\"contributor_prefix\\", \\"contributor_suffix\\", \\"contributor_street_1\\", \\"contributor_street_2\\", \\"contributor_city\\", \\"contributor_state\\", \\"contributor_zip_code\\", \\"election_code\\", \\"election_other_description\\", \\"contribution_date\\", \\"contribution_amount\\", \\"contribution_aggregate\\", \\"contribution_purpose_descrip\\", \\"contributor_employer\\", \\"contributor_occupation\\", \\"donor_committee_fec_id\\", \\"donor_committee_name\\", \\"donor_candidate_fec_id\\", \\"donor_candidate_last_name\\", \\"donor_candidate_first_name\\", \\"donor_candidate_middle_name\\", \\"donor_candidate_prefix\\", \\"donor_candidate_suffix\\", \\"donor_candidate_office\\", \\"donor_candidate_state\\", \\"donor_candidate_district\\", \\"conduit_name\\", \\"conduit_street1\\", \\"conduit_street2\\", \\"conduit_city\\", \\"conduit_state\\", \\"conduit_zip_code\\", \\"memo_code\\", \\"memo_text_description\\", \\"reference_code\\", \\"filing_id\\") FROM STDIN;
",
  "SA21	C00431171	SA21.001	\\\\N	\\\\N	ORG	FLS CONNECT, LLC	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	7300 HUDSON BLVD	STE 270	SAINT PAUL	MN	55128	G2012	\\\\N	20141203	1640.25	40472.19	LIST RENTAL	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	  	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	FAIR MARKET VALUE	\\\\N	\\\\N
",
  "SA21	C00431171	SA21.002	\\\\N	\\\\N	ORG	FLS CONNECT, LLC	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	7300 HUDSON BLVD	STE 270	SAINT PAUL	MN	55128	G2012	\\\\N	20141203	3667.75	40472.19	LIST RENTAL	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	  	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	FAIR MARKET VALUE	\\\\N	\\\\N
",
  "SA21	C00431171	SA21.003	\\\\N	\\\\N	ORG	NEWSMAX MEDIA	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	560 VILLAGE BLVD.	STE 120	WEST PALM BEACH	FL	33409	G2012	\\\\N	20141203	7284.00	68171.27	LIST RENTAL	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	  	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	FAIR MARKET VALUE	\\\\N	\\\\N
",
  "\\\\.
",
  "COPY fec_expenditures (\\"form_type\\", \\"filer_committee_id_number\\", \\"transaction_id_number\\", \\"back_reference_tran_id_number\\", \\"back_reference_sched_name\\", \\"entity_type\\", \\"payee_organization_name\\", \\"payee_last_name\\", \\"payee_first_name\\", \\"payee_middle_name\\", \\"payee_prefix\\", \\"payee_suffix\\", \\"payee_street_1\\", \\"payee_street_2\\", \\"payee_city\\", \\"payee_state\\", \\"payee_zip_code\\", \\"election_code\\", \\"election_other_description\\", \\"expenditure_date\\", \\"expenditure_amount\\", \\"semi_annual_refunded_bundled_amt\\", \\"expenditure_purpose_descrip\\", \\"category_code\\", \\"beneficiary_committee_fec_id\\", \\"beneficiary_committee_name\\", \\"beneficiary_candidate_fec_id\\", \\"beneficiary_candidate_last_name\\", \\"beneficiary_candidate_first_name\\", \\"beneficiary_candidate_middle_name\\", \\"beneficiary_candidate_prefix\\", \\"beneficiary_candidate_suffix\\", \\"beneficiary_candidate_office\\", \\"beneficiary_candidate_state\\", \\"beneficiary_candidate_district\\", \\"conduit_name\\", \\"conduit_street_1\\", \\"conduit_street_2\\", \\"conduit_city\\", \\"conduit_state\\", \\"conduit_zip_code\\", \\"memo_code\\", \\"memo_text_description\\", \\"reference_to_si_or_sl_system_code_that_identifies_the_account\\", \\"filing_id\\") FROM STDIN;
",
  "SB23	C00431171	SB23.013	\\\\N	\\\\N	ORG	AIR CHARTER	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	4151 N. MULBERRY DR.	SUITE 250	KANSAS CITY	MO	64116	\\\\N	\\\\N	20141210	5740.80	\\\\N	TRAVEL: AIR	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	  	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	IN-KIND CONTRIBUTION: SASSE FOR US SENATE; SEE POST-GEN TRAN ID SB29.05	\\\\N	\\\\N
",
  "SB23	C00431171	SB23.004	\\\\N	\\\\N	ORG	ATCHLEY & ASSOCIATES	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	6850 AUSTIN CENTER BOULEVARD	SUITE 180	AUSTIN	TX	78731	\\\\N	\\\\N	20141222	299.00	\\\\N	ACCOUNTING SERVICES	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	  	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N
",
  "SB23	C00431171	SB23.005	\\\\N	\\\\N	ORG	CMDI	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	1593 SPRING HILL ROAD	SUITE 450	TYSONS CORNER	VA	22182	\\\\N	\\\\N	20141222	27500.00	\\\\N	DATA MANAGEMENT SERVICES	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	  	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N	\\\\N
",
  "\\\\.
",
]
`;
